modules/tinymce/src/core/main/ts/selection/DetailsElement.ts

```diff
@@ -11,6 +12,23 @@ const preventSummaryToggle = (editor: Editor): void => {
   });
 };
 
+const isHeading = (node: AstNode) => [ 'h1', 'h2', 'h3', 'h4', 'h5', 'h6' ].includes(node.name);
+
+// If the `summary` element contains multiple heading elements,
+// merge their contents into the first heading element.
+const handleSummaryHeadings = (details: AstNode) =>
+  Arr.findMap(details.children(), (node: AstNode) => Optionals.someIf(node.name === 'summary', node))
```

Comment:
Isn't this the same as also maybe move the summary predicate to the `AstNodeType` if you create such a module.

```suggestion
  Arr.find(details.children(), (node) => node.name === 'summary'))
```